modem/design.md

```diff
@@ -0,0 +1,8 @@
+Overview:
+
+![Design diagram](diagram.png)
+
+We create a Transceiver object that will manage the serial connection. It will do command detection and hand off to the appropriate Command object to handle the interaction. The Command object will be subclassed for each type of command we will receive and will handle the processing of the command and generating the response.
```

Comment:
what do you think of `Protocol` instead of transceiver, with the following instance
`SerialProtocol`
`SPIProtocol`
`I2CProtocol`.